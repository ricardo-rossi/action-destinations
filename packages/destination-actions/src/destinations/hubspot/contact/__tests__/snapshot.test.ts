import { createTestEvent, createTestIntegration } from '@segment/actions-core'
import { generateTestData } from '../../../../lib/test-data'
import destination from '../../index'
import nock from 'nock'
import { TransactionContext } from '@segment/actions-core/src/destination-kit'

const testDestination = createTestIntegration(destination)
const actionSlug = 'contact'
const destinationSlug = 'Hubspot'
const seedName = `${destinationSlug}#${actionSlug}`

beforeEach(() => nock.cleanAll())

describe(`Testing snapshot for ${destinationSlug}'s ${actionSlug} destination action:`, () => {
  it('required fields', async () => {
    const action = destination.actions[actionSlug]
    const [eventData, settingsData] = generateTestData(seedName, destination, action, true)

    nock(/.*/).persist().patch(/.*/).reply(200)

    const event = createTestEvent({
      properties: eventData
    })

    const transactionContext: TransactionContext = {
      transaction: {},
      setTransaction: (key, value) => ({ [key]: value })
    }

    const responses = await testDestination.testAction(actionSlug, {
      event: event,
      mapping: event.properties,
      settings: settingsData,
      auth: undefined,
      transactionContext
    })

    const request = responses[0].request
    const rawBody = await request.text()

    try {
      const json = JSON.parse(rawBody)
      expect(json).toMatchSnapshot()
      return
    } catch (err) {
      expect(rawBody).toMatchSnapshot()
    }

    expect(request.headers).toMatchSnapshot()
  })

  it('all fields', async () => {
    const action = destination.actions[actionSlug]
    const [eventData, settingsData] = generateTestData(seedName, destination, action, false)

    nock(/.*/).persist().post(/.*/).reply(200)
    nock(/.*/)
      .persist()
      .patch(/.*/)
      .reply(200, {
        id: '801',
        properties: {
          lifecyclestage: eventData.lifecyclestage.toLowerCase()
        }
      })

    const event = createTestEvent({
      properties: eventData
    })

    const transactionContext: TransactionContext = {
      transaction: {},
      setTransaction: (key, value) => ({ [key]: value })
    }

    const responses = await testDestination.testAction(actionSlug, {
      event: event,
      mapping: event.properties,
      settings: settingsData,
      auth: undefined,
      transactionContext
    })

    const request = responses[0].request
    const rawBody = await request.text()

    try {
      const json = JSON.parse(rawBody)
      expect(json).toMatchSnapshot()
      return
    } catch (err) {
      expect(rawBody).toMatchSnapshot()
    }
  })
})
