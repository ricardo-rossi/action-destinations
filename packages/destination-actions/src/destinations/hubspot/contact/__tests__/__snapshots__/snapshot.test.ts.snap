// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing snapshot for Hubspot's contact destination action: all fields 1`] = `
Object {
  "properties": Object {
    "address": "Tju6AP&lazvq)F",
    "city": "Tju6AP&lazvq)F",
    "company": "Tju6AP&lazvq)F",
    "country": "Tju6AP&lazvq)F",
    "firstname": "Tju6AP&lazvq)F",
    "lastname": "Tju6AP&lazvq)F",
    "lifecyclestage": "tju6ap&lazvq)f",
    "phone": "Tju6AP&lazvq)F",
    "state": "Tju6AP&lazvq)F",
    "testType": "Tju6AP&lazvq)F",
    "zip": "Tju6AP&lazvq)F",
  },
}
`;

exports[`Testing snapshot for Hubspot's contact destination action: required fields 1`] = `
Object {
  "properties": Object {},
}
`;
